# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContextValidate.cpp
# Commit: 904012057423
# Full Hash: 904012057423b23bf9a382da27a40869c6c00301
# Author: Ehsan Akhgari <ehsan@mozilla.com>
# Date: 2019-03-06 09:56:17
# Regressor Bug: 1532414
# File Overlap Count: 1
# Description:
#   Bug 1532414 - Pass a document/principal pointer to nsContentUtils::ShouldResistFingerprinting() callers in the canvas API; r=baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22002
# ==============================================================================

diff -r 152b9d59cb29 -r 904012057423 dom/canvas/WebGLContextValidate.cpp
--- a/dom/canvas/WebGLContextValidate.cpp	Tue Mar 05 23:13:23 2019 +0000
+++ b/dom/canvas/WebGLContextValidate.cpp	Tue Mar 05 23:20:08 2019 +0000
@@ -471,6 +471,14 @@
   gl->fGetFloatv(driverPName, mGLAliasedPointSizeRange);
 
   ////////////////
+  bool shouldResistFingerprinting =
+      mCanvasElement ?
+                     // If we're constructed from a canvas element
+          nsContentUtils::ShouldResistFingerprinting(GetOwnerDoc())
+                     :
+                     // If we're constructed from an offscreen canvas
+          nsContentUtils::ShouldResistFingerprinting(
+              mOffscreenCanvas->GetOwnerGlobal()->PrincipalOrNull());
 
   if (gfxPrefs::WebGLMinCapabilityMode()) {
     bool ok = true;
@@ -505,7 +513,7 @@
     }
 
     mDisableFragHighP = true;
-  } else if (nsContentUtils::ShouldResistFingerprinting()) {
+  } else if (shouldResistFingerprinting) {
     bool ok = true;
 
     ok &= RestrictCap(&mGLMaxTextureSize, kCommonMaxTextureSize);
